### DISCUSSION TITLE
Encryption and privacy concern on the multitenant platform

### DISCUSSION THREAD

**Discussion by nothing0012** [pockethost/pockethost#143] [Q&A]
With backup/restore from CLI in progress, wondering if there's any privacy policy and encryption (at rest) on the roadmap?

**Comment by benallfree** [comment]
Hi @nothing0012, I would welcome a PR for a privacy policy draft. If you think you can, please do.

Regarding encryption, I'm not sure I understand your question. Are you concerned that other users might be able to access your data?

**Reply by nothing0012** [reply]
ty, basically how the sqlite db is stored securely together with others in the same multitenant strucutre of pockethost, does the operator of the pockethost have direct access to all the sqlitedb without encryption?

**Reply by benallfree** [reply]
@nothing0012 Users cannot access each others data. PocketBase runs each user in a separate isolated process and only the database owned by that PocketBase instance is accessible.

I think PocketBase does not support at-rest encryption. @ganigeorgiev can you confirm that?

PocketHost sysadmins (or more accurately, the linux user account that runs the PocketBase instance) DOES have access to the databases just as it would if you hosted it yourself. 

Digital Ocean, the infrastructure provider, DOES NOT have access to the volume data because they encrypt everything.

**Reply by ganigeorgiev** [reply]
@benallfree

> I think PocketBase does not support at-rest encryption. @ganigeorgiev can you confirm that?

That's correct, there is no at-rest encryption for the db file.
Currently only the serialized settings eventually can be encrypted if `--encryptionEnv` is set (the related encryption env variable must contains a 32 chars string; for more info and example you can check https://pocketbase.io/docs/going-to-production/#enable-settings-encryption).

**Reply by benallfree** [reply]
@ganigeorgiev Can you think of any way a user could use the PocketBase REST API to access someone else's database? 

`pocketbase serve --dir=/path/to/instance/pb_data` is basically how we launch the instance and we use the binaries built by your automated release system.

**Reply by ganigeorgiev** [reply]
@benallfree No, it shouldn't be possible to access outside the `pb_data` with the existing REST APIs.
The only way I can think of is if users are allowed to use custom binaries and/or can bind to the app hooks with an interpreter like goja since then they will have access to the underlying filesystem.

Eventually, you can also try experimenting enforcing restriction on os level of the `pocketbase` process to only its `pb_data` directory (eg. [systemd](https://manpages.ubuntu.com/manpages/bionic/man5/systemd.exec.5.html) has `ReadWritePaths=`, `ReadOnlyPaths=`, etc. directives that could help with that).

**Reply by benallfree** [reply]
@ganigeorgiev I'm not planning to allow custom binaries or goja extensions at this time. 

I'm working on a Deno worker that has network isolation and can only talk to the instance associated with it. Then users can use the JS SDK to write daemons. Maybe I'll make an HTTP API handler or some other way to send messages to the Deno process. But actually, creating a `rpc` table and writing to it has worked really well in testing. The frontend client writes an rpc message to the `rpc` table, then the Deno worker is listening for realtime events and responds to them. It actually works amazingly well. 

I had considered running each pocketbase instance in a different user account, but it seems unnecessary since custom binaries/goja aren't allowed anyway.

**Comment by bmingles** [comment]
> PocketHost sysadmins (or more accurately, the linux user account that runs the PocketBase instance) DOES have access to the databases just as it would if you hosted it yourself.

I think the important distinction here would be that the sysadmin for self hosting would be ME as opposed to pockethost sysadmins who have access to the sysadmins account?

>  “just as it would if you hosted yourself” 

seems to imply it’s just as safe as self hosting. My concern here would be that a sysadmin I don’t know has access to unencrypted data? Am I misunderstanding anything? 

I really like pockethost BTW, this is the one thing that concerns me for storing user data that is not mine.
